torch/_dynamo/variables/builder.py

```diff
@@ -1119,6 +1119,88 @@ def __eq__(self, other: object) -> bool:
         return False
 
 
+def _automatic_dynamic(e, tx, name, static_shapes):
```

Comment:
all this shit was factored out from `wrap_to_fake_tensor_and_record()`; no changes

Reply:
We're modifying this code a lot, recommend carving this into its own PR and landing ASAP.

Reply:
#101114